packages/matter.js/src/node/server/TransactionalInteractionServer.ts

```diff
@@ -119,18 +153,71 @@ export class TransactionalInteractionServer extends InteractionServer {
             session: exchange.session,
             tracer: this.#tracer,
             actionType: ActionTracer.ActionType.Read,
+            endpoint,
+            root: this.#endpoint,
         }).act(readAttribute);
     }
 
+    protected override async readEvent(
+        path: EventPath,
+        eventFilters: TypeFromSchema<typeof TlvEventFilter>[] | undefined,
+        event: EventServer<any, any>,
+        exchange: MessageExchange<MatterDevice>,
+        fabricFiltered: boolean,
+        message: Message,
+        endpoint: EndpointInterface,
+    ): Promise<EventStorageData<any>[]> {
+        const readEvent = (context: ActionContext) => {
+            // TODO Change this to behavior style, right now it does minimalistic whats needed
```

Comment:
What would "behavior style" look like?  Extend behavior API to support reading historical events, then doing ACL check there?

Reply:
Good question. If you are ok with thatcher all good. ;-)